devito/data/data.py

```diff
@@ -155,12 +155,24 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def _prune_shape(self, shape):
```

Comment:
_reduce_eq_shape?

Reply:
It's not an equation, just a `Data` object, but I'm open to alternative names.